src/libraries/System.Private.CoreLib/src/System/IO/PathInternal.cs

```diff
@@ -102,40 +102,90 @@ internal static bool AreRootsEqual(string? first, string? second, StringComparis
         }
 
         /// <summary>
-        /// Try to remove relative segments from the given path (without combining with a root).
+        /// Tries to remove relative segments from the given path, starting after the path root.
         /// </summary>
-        /// <param name="path">Input path</param>
-        /// <param name="rootLength">The length of the root of the given path</param>
-        internal static string RemoveRelativeSegments(string path, int rootLength)
+        /// <param name="path">The input path.</param>
+        /// <returns>The path without relative segments.</returns>
+        internal static string RemoveRelativeSegments(string path)
         {
+            Debug.Assert(path != null);
+            Debug.Assert(path.Length > 0);
+
+            return RemoveRelativeSegments(path.AsSpan());
+        }
+
+        /// <summary>
+        /// Tries to remove relative segments from the given path, starting after the path root.
+        /// </summary>
+        /// <param name="path">The input path.</param>
+        /// <returns>The path without relative segments.</returns>
+        internal static string RemoveRelativeSegments(ReadOnlySpan<char> path)
+        {
+            Debug.Assert(path.Length > 0);
+
+            ReadOnlySpan<char> pathRoot = Path.GetPathRoot(path);
+
+            return RemoveRelativeSegments(path, pathRoot.Length);
+        }
+
+        /// <summary>
+        /// Tries to remove relative segments from the given path, starting the analysis at the specified location.
+        /// </summary>
+        /// <param name="path">The input path.</param>
+        /// <param name="charactersToSkip">The number of characters to ignore from the beginning of the path string.</param>
+        /// <returns>The path without relative segments.</returns>
+        internal static string RemoveRelativeSegments(string path, int charactersToSkip)
+        {
+            Debug.Assert(path != null);
+            Debug.Assert(path.Length > 0);
+            Debug.Assert(charactersToSkip > 0);
```

Comment:
What guarantees that `charactersToSkip` is `> 0`? `GetPathRoot` can return empty string.